### PR TITLE
Add API endpoint

### PR DESCRIPTION
**Pull request by MarcoDiaz2000**
## Description
In this project, we add some API endpoints to our Blog app. This will allow our app to be used in different ways, rather than just through your site - for example, a mobile app or a CLI application.

### Project requirements

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def create
+    post = Post.find(params[:post_id])
+    comment = post.comments.new(comment_params)
+    comment.author_id = current_user.id
+
+    if comment.save
+      render json: { notice: 'Comment successfully created', comment: }, status: :created
+    else
+      render json: { errors: comment.errors.full_messages }, status: :unprocessable_entity
+    end
```
Comment (christianbiring1) [inline app/controllers/api/comments_controller.rb]: - [x] Kindly make sure that you have the correct file and folder structure in your Rails application. In a versioned API, you would typically have a directory structure like this:
`app/controllers/api/v1/`
```rb
# Inside the v1 directory, you should have your API controllers and modules.
```
image

File: config/routes.rb [hunk 2]
```diff
@@ -13,5 +13,18 @@
       resources :likes, only: [:create, :destroy]
     end
   end
```
Comment (christianbiring1) [inline config/routes.rb:15]: [OPTIONAL]
- [x] Also make sure you don't have the `views` endpoints as they it's not necessary to have them in a `rails backend application`

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @MarcoDiaz2000 @tumainimaganiko , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUIRED CHANGES :recycle: 

## Highlights
- [x] Great work for defining your end-point controllers properly for posts and comments :100: 
- [x] Work well documented :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Please make sure you test your end-point and make sure are working properly before you submit :100: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/comments_controller.rb (hunk 1 above)
Comment (MarcoDiaz2000) [reply app/controllers/api/comments_controller.rb:16]: Hi @christianbiring1 
thanks for the comments.

in that order we have it but it seems that when it was uploaded it was changed
api

Comment (MarcoDiaz2000) [reply app/controllers/api/comments_controller.rb]: I will fix it, thanks

**Review by souad988 — APPROVED** [review]
Hi @MarcoDiaz2000 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

approved2


## Highlights
Good communication
Document the work 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarcoDiaz2000** [comment]
Thank you very much.

**Pull request merged**